src/dotty/tools/dotc/typer/TypeAssigner.scala

```diff
@@ -470,7 +470,7 @@ trait TypeAssigner {
     tree.withType(sym.nonMemberTermRef)
 
   def assignType(tree: untpd.Annotated, annot: Tree, arg: Tree)(implicit ctx: Context) =
-    tree.withType(AnnotatedType(arg.tpe, Annotation(annot)))
+    tree.withType(AnnotatedType(arg.tpe.widen, Annotation(annot)))
```

Comment:
I'm not sure if this is a good change. It makes ` val a : 1 = (1 : @annotation)` not typecheck as type of `1` will get widened to Int.